packages/custom-objects-importer/src/main.js

```diff
@@ -0,0 +1,295 @@
+            msg = 'Create error occurred but ignored. See summary for details'
+          }
+          this.logger.error(msg)
+          return Promise.resolve()
+        }
+        if (update) {
+          this._summary.updateErrorCount += 1
+          msg =
+            'Process stopped due to error while updating custom object. See summary for details'
+        } else {
+          this._summary.createErrorCount += 1
+          msg =
+            'Process stopped due to error while creating custom object. See summary for details'
+        }
+        this.logger.error(msg)
+
+        return Promise.reject(
+          new CustomObjectImportError(
+            msg,
+            this._summary,
+            error.message || error
+          )
+        )
+      })
+  }
+
+  _createOrUpdateObjects(
+    existingObjects: Array<CustomObject>,
+    newObjects: Array<CustomObjectDraft>
+  ): Array<ClientRequest> {
+    this.logger.info('Executing...')
+
+    const createOrUpdateObjects = newObjects.map(
+      (newObject: CustomObjectDraft): CustomObjectDraft | null => {
+        const existing = existingObjects.find(
+          (oldObject: CustomObject): boolean =>
+            oldObject.key === newObject.key &&
+            oldObject.container === newObject.container
+        )
+        if (existing) {
+          if (isEqual(newObject.value, existing.value)) {
+            this._summary.unchanged += 1
+            return null
+          }
+          return { ...newObject, update: true }
+        }
+
+        return newObject
+      }
+    )
+
+    return this._buildRequests(compact(createOrUpdateObjects))
+  }
+
+  _buildRequests(newObjects: Array<CustomObjectDraft>): Array<ClientRequest> {
+    this.logger.info('Creating requests...')
+    const uri = CustomObjectsImporter.buildUri(this.apiConfig.projectKey)
+
+    return newObjects.map((newObject: CustomObjectDraft): ClientRequest =>
+      CustomObjectsImporter.buildRequest(uri, 'POST', newObject)
+    )
+  }
+
+  static buildUri(projectKey: string): string {
+    const service = createRequestBuilder({
+      projectKey,
+    }).customObjects
+    return service.build()
+  }
+
+  static buildRequest(
+    uri: string,
+    method: MethodType,
+    body?: string | Object
+  ): ClientRequest {
+    return body ? { uri, method, body } : { uri, method }
+  }
+
+  _initiateSummary(): Summary {
+    this._summary = {
+      created: 0,
+      updated: 0,
+      unchanged: 0,
+      createErrorCount: 0,
+      updateErrorCount: 0,
+      errors: [],
+    }
+    return this._summary
+  }
+
+  summaryReport(): SummaryReport {
+    const {
+      created,
+      updated,
+      unchanged,
+      createErrorCount,
+      updateErrorCount,
+    } = this._summary
+    let message
+    if (!created && !updated && !createErrorCount && !updateErrorCount)
+      message = 'Summary: nothing to do, all objects are left unchanged'
+    else
+      message = `Summary: there were ${created +
```

Comment:
I think we still use `common-tags` among our packages...
let's use that?
ref: https://github.com/declandewet/common-tags

Reply:
fixed [here](https://github.com/commercetools/nodejs/pull/651/commits/4ef42ff9ae4cd49886b0c9e979c8872ed21ee8af#diff-65284eab390612879ea62171c18094a3R281)